comictaggerlib/ui/talkeruigenerator.py

```diff
@@ -0,0 +1,225 @@
+    row = layout.rowCount()
+    lbl = QtWidgets.QLabel(option.display_name)
+    layout.addWidget(lbl, row, 0)
+    widget = QtWidgets.QDoubleSpinBox()
+    widget.setRange(0, 9999.99)
+    widget.setToolTip(option.help)
+    layout.addWidget(widget, row, 1, alignment=QtCore.Qt.AlignLeft)
+
+    return widget
+
+
+def generate_textbox(option: settngs.Setting, layout: QtWidgets.QGridLayout) -> QtWidgets.QLineEdit:
+    row = layout.rowCount()
+    lbl = QtWidgets.QLabel(option.display_name)
+    layout.addWidget(lbl, row, 0)
+    widget = QtWidgets.QLineEdit()
+    lbl.setToolTip(option.help)
+    widget.setToolTip(option.help)
+    layout.addWidget(widget, row, 1)
+
+    return widget
+
+
+def settings_to_talker_form(sources: dict[str, QtWidgets.QWidget], config: settngs.Config[settngs.Namespace]) -> None:
+    # Set the active talker via id in sources combo box
+    sources["cbx_select_talker"].setCurrentIndex(sources["cbx_select_talker"].findData(config[0].talker_source))
+
+    for talker in sources["tabs"].items():
+        for name, widget in talker[1]["widgets"].items():
+            value = getattr(config[0], name)
+            value_type = type(value)
+            try:
+                if value_type is str:
+                    widget.setText(value)
+                if value_type is int or value_type is float:
+                    widget.setValue(value)
+                if value_type is bool:
+                    widget.setChecked(value)
+            except Exception:
+                logger.debug("Failed to set value of %s", name)
+
+
+def form_settings_to_config(sources: dict[str, QtWidgets.QWidget], config: settngs.Config[settngs.Namespace]) -> None:
+    # Source combo box value
+    config[0].talker_source = sources["cbx_select_talker"].currentData()
+
+    for tab in sources["tabs"].items():
+        for name, widget in tab[1]["widgets"].items():
+            widget_value = None
+            if isinstance(widget, (QtWidgets.QSpinBox, QtWidgets.QDoubleSpinBox)):
+                widget_value = widget.value()
+            elif isinstance(widget, QtWidgets.QLineEdit):
+                widget_value = widget.text().strip()
+            elif isinstance(widget, QtWidgets.QCheckBox):
+                widget_value = widget.isChecked()
+
+            setattr(config[0], name, widget_value)
+
+
+def generate_source_option_tabs(
+    comic_talker_tab: QtWidgets.QWidget,
+    config: settngs.Config[settngs.Namespace],
+    talkers: dict[str, ComicTalker],
+) -> dict[str, QtWidgets.QWidget]:
+    """
+    Generate GUI tabs and settings for talkers
+    """
+
+    # Store all widgets as to allow easier access to their values vs. using findChildren etc. on the tab widget
+    sources: dict = {"tabs": {}}
+
+    # Tab comes with a QVBoxLayout
+    comic_talker_tab_layout = comic_talker_tab.layout()
+
+    talker_layout = QtWidgets.QGridLayout()
+    lbl_select_talker = QtWidgets.QLabel("Metadata Sources:")
```

Comment:
This needs to indicate that it is the currently selected talker at a minimum change it back to metadata source

```suggestion
    lbl_select_talker = QtWidgets.QLabel("Metadata Source:")
```